pkg/operator/kubeclient/kube_client_container.go

```diff
@@ -0,0 +1,58 @@
+package kubeclient
+
+import (
+	certmanoperatorclient "github.com/openshift/cert-manager-operator/pkg/operator/clientset/versioned"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type KubeClientContainer struct {
+	KubeConfig                *kubernetes.Clientset
+	DynamicClient             dynamic.Interface
+	ConfigClient              *configv1client.Clientset
+	CertManagerOperatorClient *certmanoperatorclient.Clientset
+	ApiExtensionsClient       *apiextensionsclient.Clientset
+}
+
+func NewKubeClientContainer(protoKubeConfig *rest.Config, kubeConfig *rest.Config) (KubeClientContainer, error) {
```

Comment:
Use the clients directly rather than bundled in order to maintain better outlook about what might be happening inside of each of the controllers.

Reply:
This doesn't make sense to be honest as most of the controllers hide their intention by using `ClientHolder`. The problem with `ClientHolder` is that is has no public accessors, so it can not be used anywhere else except library-go.

Reply:
Since we're using the `staticResourceController` now, you can just use `resourceapply.NewKubeClientHolder(kubeClient)` and can throw this all away

Reply:
I like this idea! Let me put it into life.

Reply:
Unfortunately, it turned out I can't do it. In certain places, I need ConfigClient (`configv1client.NewForConfig(kubeConfig)`) and Cert Manager Operator Client (`certmanoperatorclient.NewForConfig(kubeConfig)`). The `ClientHolder` construct doesn't give me encapsulation I need there. 

I'm leaving this as is.

Reply:
This shows you how to get rid of your custom holder + does a bunch of code clean up: https://github.com/stlaz/cert-manager-operator/commit/838e762b55a151cd92526fcca422377d4914eeac. Please apply on this commit.